packages/edit-site/src/components/editor/use-resolve-edited-entity.js

```diff
@@ -32,33 +32,14 @@ const authorizedPostTypes = [ 'page', 'post' ];
 export function useResolveEditedEntity() {
 	const { params = {} } = useLocation();
 	const { postId, postType } = params;
-	const { hasLoadedAllDependencies, homepageId, postsPageId } = useSelect(
-		( select ) => {
-			const { getEntityRecord } = select( coreDataStore );
-			const siteData = getEntityRecord( 'root', 'site' );
-			const _homepageId =
-				siteData?.show_on_front === 'page' &&
-				[ 'number', 'string' ].includes(
-					typeof siteData.page_on_front
-				) &&
-				!! +siteData.page_on_front // We also need to check if it's not zero(`0`).
-					? siteData.page_on_front.toString()
-					: null;
-			const _postsPageId =
-				siteData?.show_on_front === 'page' &&
-				[ 'number', 'string' ].includes(
-					typeof siteData.page_for_posts
-				)
-					? siteData.page_for_posts.toString()
-					: null;
-			return {
-				hasLoadedAllDependencies: !! siteData,
-				homepageId: _homepageId,
-				postsPageId: _postsPageId,
-			};
-		},
-		[]
-	);
+	const { homepagePostType, homepagePostId } = useSelect( ( select ) => {
+		const { getHomePage } = unlock( select( coreDataStore ) );
+		const homePage = getHomePage();
+		return {
+			homepagePostType: homePage?.postType,
+			homepagePostId: homePage?.postId,
+		};
+	}, [] );
```

Comment:
Could you just return `homePage` from the `useSelect` instead of creating an extra object?

Reply:
If I do I have to memoize the getHomePage selector and I was lazy :P

Reply:
I memoized the selector in [c33d560](https://github.com/WordPress/gutenberg/pull/67031/commits/c33d5601169a024bb28d74f6d900ca5f0ed062ac) let me know if this is correct